### ISSUE TITLE
Certain products are hidden on the order page

### ISSUE THREAD

**Issue by joashrajin** [woocommerce/woocommerce-ios#7448]
**Describe the bug**
When choosing products to be added to an order, some aren't shown, but the price is added to the total. We've spent a lot of time troubleshooting this but could not notice anything odd. 

In the app debug logs, we notice instances of `Error: string input is not a number:` but don't see which field this is applying to when accessing the affected products which are hidden. I can replicate this when importing these variable products too, but couldn't locate the cause of the problem. 

Ticket with all of the information: #5463554-zen

**To Reproduce**
1. Try adding a variable product to an order
2. The product will not show up but the price is included

**Screen Recording**

There are two screen recordings. One in #5463554-zen and the other is linked in the comment below (since it's after they disabled plugins). They show some private information so I'm just sharing links.

**Expected behavior**
The product should show up on the order page

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: iPhone 13
- iOS version: 15.6
- WooCommerce iOS version: 9.8

**WordPress Environment**

<details>

```
### WordPress Environment ###

WordPress address (URL): removed
Site address (URL): -removed
WC Version: 6.7.0
REST API Version: ✔ 6.7.0
WC Blocks Version: ✔ 7.8.3
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.4.30
PHP Post Max Size: 32 MB
PHP Time Limit: 90
PHP Max Input Vars: 3000
cURL Version: 7.84.0
OpenSSL/1.1.1q

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.35-MariaDB
Max Upload Size: 32 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.7.0
WC Database Prefix: wp_
Total Database Size: 149.12MB
Database Data Size: 133.20MB
Database Index Size: 15.92MB
wp_woocommerce_sessions: Data: 1.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.22MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 2.50MB + Index: 2.97MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 3.31MB + Index: 2.02MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 2.38MB + Index: 1.88MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 1.52MB + Index: 0.45MB + Engine InnoDB
wp_e_events: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_e_notes: Data: 0.02MB + Index: 0.17MB + Engine InnoDB
wp_e_notes_users_relations: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_e_submissions: Data: 0.06MB + Index: 0.27MB + Engine InnoDB
wp_e_submissions_actions_log: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_e_submissions_values: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_iwp_backup_status: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_iwp_file_list: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_iwp_processed_iterator: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_jet_post_types: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_jet_smart_filters_indexer: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_jet_taxonomies: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 50.02MB + Index: 0.44MB + Engine InnoDB
wp_postmeta: Data: 33.11MB + Index: 3.95MB + Engine InnoDB
wp_posts: Data: 14.05MB + Index: 0.39MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 1.47MB + Index: 0.83MB + Engine InnoDB
wp_users: Data: 0.08MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.22MB + Index: 0.25MB + Engine InnoDB
wp_wc_order_stats: Data: 0.05MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.05MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfblockediplog: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfblocks7: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfconfig: Data: 0.47MB + Index: 0.00MB + Engine InnoDB
wp_wfcrawlers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilechanges: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilemods: Data: 12.55MB + Index: 0.00MB + Engine InnoDB
wp_wfhits: Data: 1.47MB + Index: 0.20MB + Engine InnoDB
wp_wfhoover: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfknownfilelist: Data: 6.52MB + Index: 0.00MB + Engine InnoDB
wp_wflivetraffichuman: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wflocs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wflogins: Data: 0.16MB + Index: 0.03MB + Engine InnoDB
wp_wfls_2fa_secrets: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfls_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfnotifications: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfpendingissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfreversecache: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfsnipcache: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfstatus: Data: 0.13MB + Index: 0.11MB + Engine InnoDB
wp_wftrafficrates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_debug_events: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmelon_advbedit_temp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yoast_indexable: Data: 0.42MB + Index: 0.31MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.06MB + Index: 0.03MB + Engine InnoDB

### Post Type Counts ###

attachment: 251
csp_rules: 2
elementor_font: 1
elementor_library: 15
iwp_log: 20
jet-engine-booking: 1
jet-menu: 1
jet-smart-filters: 3
nav_menu_item: 7
oembed_cache: 1
page: 30
post: 2
product: 38
product_variation: 160
revision: 1011
shop_order: 281
wp_global_styles: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (34) ###

InfiniteWP - Client: by Revmakx – 1.9.8
weLaunch Framework: by weLaunch.io – 1.0.12
Duplicate Page: by mndpsingh287 – 4.4.9
Elementor Pro: by Elementor.com – 3.7.3
Elementor: by Elementor.com – 3.7.0
Facebook for WooCommerce: by Facebook – 2.6.19
JetBlocks For Elementor: by Crocoblock – 1.3.2
JetElements For Elementor: by Crocoblock – 2.6.6
JetEngine: by Crocoblock – 3.0.2
JetMenu: by Crocoblock – 2.2.2
JetSearch: by Crocoblock – 3.0.1
JetSmartFilters: by Crocoblock – 2.3.13
JetWooBuilder For Elementor: by Crocoblock – 2.0.2
JetProductGallery: by Crocoblock – 2.1.5
Jetpack: by Automattic – 11.2
Klaviyo: by Klaviyo
Inc. – 3.0.3

Make Column Clickable Elementor: by Fernando Acosta – 1.4.0
Really Simple SSL: by Really Simple Plugins – 5.3.3
Role Based Pricing for WooCommerce: by Addify – 1.6.0
User Role Editor: by Vladimir Garagulya – 4.63
WooCommerce Advanced Bulk Edit: by George Iron & Yas G. – 5.1
WooCommerce Conditional Shipping and Payments: by WooCommerce – 1.14.0
WooCommerce Stripe Gateway: by WooCommerce – 6.5.1
Pixel Manager for WooCommerce: by SweetCode – 1.18.1
WooCommerce Google Analytics Integration: by WooCommerce – 1.5.13
WooCommerce Shipping & Tax: by WooCommerce – 1.26.3
WooCommerce USPS Shipping: by WooCommerce – 4.4.74
WooCommerce - ShipStation Integration: by WooCommerce – 4.1.48
WooCommerce Variations Table: by weLaunch – 1.3.10
WooCommerce: by Automattic – 6.7.0
Wordfence Security: by Wordfence – 7.6.0
Yoast SEO: by Team Yoast – 19.4
WP Mail SMTP: by WPForms – 3.5.1
YITH Pre-Order for WooCommerce Premium: by YITH – 2.3.0

### Inactive Plugins (6) ###

Multi-Carrier ShipStation Shipping for WooCommerce: by OneTeamSoftware – 1.4.7
Thinkific Uploader: by Thinkific – 1.0.0
USPS Live Rates: by Octolize – 1.2.0
WooCommerce Payments: by Automattic – 4.5.0
WooCommerce Pre-Orders: by WooCommerce – 1.7.2
WooCommerce Wholesale Prices: by Rymera Web Co – 2.1.4

### Must Use Plugins (1) ###

InfiniteWP - Client Loader: by Revmakx – 1.0.1

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔

### WC Pages ###

Shop base: #368 - /shop/
Cart: 
Checkout: 
 block.
My account: #371 - /my-account/
Terms and conditions: #297 - /terms-of-use/

### Theme ###

Name: Hello Elementor Child
Version: 1.0.1
Author URL: https://elementor.com/
Child Theme: ✔
Parent Theme Name: Hello Elementor
Parent Theme Version: 2.6.1
Parent Theme Author URL: https://elementor.com/?utm_source=wp-themes&utm_campaign=author-uri&utm_medium=wp-dash
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
experimental-product-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
wc-pay-promotion
wc-pay-welcome-page
wc-pay-subscriptions-page

Disabled Features: minified-js
settings

Daily Cron: ✔ Next scheduled: 2022-08-08 23:33:36 -05:00
Options: ✔
Notes: 54
Onboarding: completed

### Action Scheduler ###

Complete: 7,329
Oldest: 2022-07-13 19:04:31 -0500
Newest: 2022-08-08 13:10:26 -0500

Pending: 3
Oldest: 2022-08-08 14:01:32 -0500
Newest: 2022-08-09 02:55:06 -0500


### Status report information ###

Generated at: 2022-08-08 13:22:10 -05:00
```
</details>

**Comment by ChrissiePollock** [comment]
The customer came back saying:  I was they tested the website with only the Storefront theme and WooCommerce installed. They shared the video/ logs of that test still showing this bug: products added to an order do not appear on the  list, but appear afterwards once the order is created. That video can be seen on #5466621-zen.

**Comment by jostnes** [comment]
I was able to reproduce this issue, the issue is intermittent but it seems like it's easier to reproduce after a sign-in:

https://user-images.githubusercontent.com/17252150/184089406-8a36e75c-810f-483e-a8ad-a04878706f02.MP4


cc: @rachelmcr @Ecarrion @ealeksandrov

**Comment by Ecarrion** [comment]
Thanks for letting us know @jostnes, @joashrajin, and @ChrissiePollock I have submitted a fix  for the problem, will let you know once it is merged.

**Comment by Ecarrion** [comment]
@joashrajin @ChrissiePollock this has been merged and is scheduled to be fixed in `release 9.9` I'm going to close this issue but feel free to reopen it in case it does not fix the merchant problem.
